src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -1070,32 +1070,26 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                 if (diagsWritten && !methodSymbol.IsImplicitlyDeclared && _compilation.EventQueue != null)
                 {
-                    Lazy<SemanticModel> lazySemanticModel = null;
-
-                    if (body != null)
+                    // If compilation has a caching semantic model provider, then cache the already-computed bound tree
+                    // onto the semantic model and store it on the event.
+                    SyntaxTreeSemanticModel semanticModelWithCachedBoundNodes = null;
```

Comment:
Why did we move from a lazy to prompt get of the semantic model here? Seems like we could have created the lazy and used the `CachingSemanticModel` inside to create the model. Is there a reason that is not desirable here?

Reply:
I wasn't sure if laziness would give us any benefit now that we have switched from weakly referenced semantic model in the event to a strongly referenced one that is backed by client owned caching semantic model provider. I can switch to lazy one and measure and perf changes if that helps, but code wise this looked cleaner to me.